src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -209,6 +213,10 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             if (Href != null)
             {
                 output.CopyHtmlAttribute(HrefAttributeName, context);
+
+                Href = UrlHelper.Content(Href.Trim());
```

Comment:
`Href` isn't used again for about a 100 lines. Don't want someone to think the assignment is unnecessary.